extensions/amp-subscriptions/0.1/amp-subscriptions.js

```diff
@@ -276,26 +272,24 @@ export class SubscriptionService {
         granted: grantState,
       };
 
-      const selectedPlatform = dev().assert(
-          this.subscriptionPlatforms_[selectedEntitlement.service],
-          'Selected service not registered');
-
       selectedPlatform.activate(renderState);
 
-      this.viewTrackerPromise_.then(() => {
-        const localPlatform = /** @type {!LocalSubscriptionPlatform} */ (
-          user().assert(this.subscriptionPlatforms_['local'],
-              'Local platform is not registered'));
-
-        if (selectedPlatform.isPingbackEnabled()) {
-          selectedPlatform.pingback(selectedEntitlement);
-        }
-
-        if (selectedPlatform.getServiceId() !== localPlatform.getServiceId()
-            && localPlatform.isPingbackEnabled()) {
-          localPlatform.pingback(selectedEntitlement);
-        }
-      });
+      if (this.viewTrackerPromise_) {
+        this.viewTrackerPromise_.then(() => {
+          const localPlatform = /** @type {!LocalSubscriptionPlatform} */ (
+            user().assert(this.platformStore_.getLocalPlatform(),
```

Comment:
I think the `user().assert()` should just go to `getLocalPlatform` implementation itself.